src/foundry/common/prosemirror/_module.d.mts

```diff
@@ -1,25 +1,30 @@
-import * as collab from "prosemirror-collab";
-import { keymap } from "prosemirror-keymap";
-import { DOMParser, DOMSerializer, Schema } from "prosemirror-model";
-import type { Plugin, PluginKey } from "prosemirror-state";
-import { AllSelection, TextSelection, EditorState } from "prosemirror-state";
-import { Step } from "prosemirror-transform";
+// In Foundry itself this file contains re-exports of these other modules.
+// Therefore it has a runtime effect and uses `.mjs` instead of `.d.mts`.
+// While `.mts` could work, to avoid `import/no-unresolved` from erroring `.mjs` is used.
+/* eslint-disable import/extensions */
```

Comment:
Could you move this comment down to right below the re-exports, i.e. right above:
```
export * as commands from "prosemirror-commands";
export * as transform from "prosemirror-transform";
export * as list from "prosemirror-schema-list";
export * as tables from "prosemirror-tables";
export * as input from "prosemirror-inputrules";
export * as state from "prosemirror-state";
```

This'll cause some lints to appear on your imports, this is the point. We want to be using `.mts` and `import type` consistently. In imports it makes it easier to bundle and makes TS do less work (in theory) etc.

On the import end I've yet to find a good reason to use `.mjs` when you have a `.mts` available. The comment I wrote is specifically about the `export x from "y"` stuff

Reply:
Should I then rework then imports to fix the lints about the package imports (i.e. switch them to be export x from y)?

And doesn't it need an mjs import to include constants declared in the mjs?

Reply:
Or should I be redeclaring those...

Reply:
> doesn't it need an mjs import to include constants declared in the mjs?

I'm not quite sure what you mean. For one our repo doesn't have `.mjs` files but I assume you mean in Foundry?

If I assume that's what you mean then the answer is simply "no". In a `.d.mts` file you can still write `declare const x: ...` and then later import it.

Try this version of the file, there no need for the disable even because the only re-exports are from an npm package.
```ts
import { EditorState, AllSelection, TextSelection, Plugin, PluginKey } from "prosemirror-state";
import { EditorView } from "prosemirror-view";
import { Schema, DOMSerializer } from "prosemirror-model";
// eslint-disable-next-line import/no-named-as-default
import type ProseMirrorInputRules from "./input-rules.d.mts";
import { keymap } from "prosemirror-keymap";
// eslint-disable-next-line import/no-named-as-default
import type ProseMirrorKeyMaps from "./keymaps.d.mts";
// eslint-disable-next-line import/no-named-as-default
import type ProseMirrorMenu from "./menu.d.mts";
import "./extensions.d.mts";
import type * as collab from "prosemirror-collab";
import type { Step } from "prosemirror-transform";
import type { parseHTMLString, serializeHTMLString } from "./util.d.mts";
import type { schema as defaultSchema } from "./schema.d.mts";
import type ProseMirrorPlugin from "./plugin.d.mts";
import type ProseMirrorImagePlugin from "./image-plugin.d.mts";
import type ProseMirrorDirtyPlugin from "./dirty-plugin.d.mts";
import type ProseMirrorContentLinkPlugin from "./content-link-plugin.d.mts";
import type ProseMirrorHighlightMatchesPlugin from "./highlight-matches-plugin.d.mts";
import type ProseMirrorClickHandler from "./click-handler.d.mts";
import type DOMParser from "./dom-parser.d.mts";

declare const dom: {
  parser: DOMParser;
  serializer: DOMSerializer;
  parseString: typeof parseHTMLString;
  serializeString: typeof serializeHTMLString;
};

declare const defaultPlugins: Record<
  | "inputRules"
  | "keyMaps"
  | "menu"
  | "isDirty"
  | "clickHandler"
  | "pasteTransformer"
  | "baseKeyMap"
  | "dropCursor"
  | "gapCursor"
  | "history"
  | "columnResizing"
  | "tables",
  Plugin
>;

export * as commands from "prosemirror-commands";
export * as transform from "prosemirror-transform";
export * as list from "prosemirror-schema-list";
export * as tables from "prosemirror-tables";
export * as input from "prosemirror-inputrules";
export * as state from "prosemirror-state";

export {
  AllSelection,
  TextSelection,
  DOMParser,
  DOMSerializer,
  EditorState,
  EditorView,
  Schema,
  Step,
  Plugin,
  PluginKey,
  ProseMirrorPlugin,
  ProseMirrorContentLinkPlugin,
  ProseMirrorHighlightMatchesPlugin,
  ProseMirrorDirtyPlugin,
  ProseMirrorImagePlugin,
  ProseMirrorClickHandler,
  ProseMirrorInputRules,
  ProseMirrorKeyMaps,
  ProseMirrorMenu,
  collab,
  defaultPlugins,
  defaultSchema,
  dom,
  keymap,
};
```

If you can find some way to break this then please let me know.

Reply:
Specifically, here's what I'm struggling with...

_module.ts does this:
```
import type { schema as defaultSchema } from "./schema.d.mts";
export { defaultSchema, ... }
```

schema.d.mts:
```
import type { MarkSpec, NodeSpec, Schema } from "prosemirror-model";
export declare const schema: Schema;
```

This test needs to pass:
```
expectTypeOf(ProseMirror.defaultSchema).toEqualTypeOf<Schema>();
```

Reply:
Alright this should work:
```
import { EditorState, AllSelection, TextSelection, Plugin, PluginKey } from "prosemirror-state";
import { EditorView } from "prosemirror-view";
import { Schema, DOMSerializer } from "prosemirror-model";
// eslint-disable-next-line import/no-named-as-default
import type ProseMirrorInputRules from "./input-rules.d.mts";
import { keymap } from "prosemirror-keymap";
// eslint-disable-next-line import/no-named-as-default
import type ProseMirrorKeyMaps from "./keymaps.d.mts";
// eslint-disable-next-line import/no-named-as-default
import type ProseMirrorMenu from "./menu.d.mts";
import "./extensions.d.mts";
import * as collab from "prosemirror-collab";
import type { Step } from "prosemirror-transform";
import type { parseHTMLString, serializeHTMLString } from "./util.d.mts";

/** A const is being imported here. */
// eslint-disable-next-line import/extensions
import { schema as defaultSchema } from "./schema.mjs";
import type ProseMirrorPlugin from "./plugin.d.mts";
import type ProseMirrorImagePlugin from "./image-plugin.d.mts";
import type ProseMirrorDirtyPlugin from "./dirty-plugin.d.mts";
import type ProseMirrorContentLinkPlugin from "./content-link-plugin.d.mts";
import type ProseMirrorHighlightMatchesPlugin from "./highlight-matches-plugin.d.mts";
import type ProseMirrorClickHandler from "./click-handler.d.mts";
import type DOMParser from "./dom-parser.d.mts";

declare const dom: {
  parser: DOMParser;
  serializer: DOMSerializer;
  parseString: typeof parseHTMLString;
  serializeString: typeof serializeHTMLString;
};

declare const defaultPlugins: Record<
  | "inputRules"
  | "keyMaps"
  | "menu"
  | "isDirty"
  | "clickHandler"
  | "pasteTransformer"
  | "baseKeyMap"
  | "dropCursor"
  | "gapCursor"
  | "history"
  | "columnResizing"
  | "tables",
  Plugin
>;

export * as commands from "prosemirror-commands";
export * as transform from "prosemirror-transform";
export * as list from "prosemirror-schema-list";
export * as tables from "prosemirror-tables";
export * as input from "prosemirror-inputrules";
export * as state from "prosemirror-state";

export {
  AllSelection,
  TextSelection,
  DOMParser,
  DOMSerializer,
  EditorState,
  EditorView,
  Schema,
  Step,
  Plugin,
  PluginKey,
  ProseMirrorPlugin,
  ProseMirrorContentLinkPlugin,
  ProseMirrorHighlightMatchesPlugin,
  ProseMirrorDirtyPlugin,
  ProseMirrorImagePlugin,
  ProseMirrorClickHandler,
  ProseMirrorInputRules,
  ProseMirrorKeyMaps,
  ProseMirrorMenu,
  collab,
  defaultPlugins,
  defaultSchema,
  dom,
  keymap,
};
```

Reply:
success